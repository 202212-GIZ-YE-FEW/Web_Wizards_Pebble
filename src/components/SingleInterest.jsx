import { Button } from "react-flatifycss";

const SingleInterest = ({ text }) => {
    const styles = {
        border: "2px solid black",
        borderRadius: "0.5rem",
        boxShadow: "2px 2px #1a1a1a",
    };

    return (
        <div className=' w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 '>
            <Button
                style={styles}
                className='bg-white w-full lg:h-18 h-16 px-1 transition focus:bg-primary-100'
            >
                <div className='relative flex items-center justify-center '>
                    <p className='block w-max font-semibold text-gray-900 text-xs py-1 '>
                        {text}
                    </p>
                </div>
            </Button>
        </div>
    );
};

export default SingleInterest;
