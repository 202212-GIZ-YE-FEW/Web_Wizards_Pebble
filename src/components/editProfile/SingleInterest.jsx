import { Button } from "react-flatifycss";
const SingleInterest = ({ text, id, clicked, onInterestClick }) => {
    return (
        <div className='w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 '>
            <Button
                className={` w-full lg:h-18 h-16 px-1 transition ${
                    clicked ? "!bg-primary-100" : "bg-white"
                } !border-2 !rounded-lg !shadow-[2px_2px_#1A1A1A]`}
                onClick={() => onInterestClick(id)}
                type='button'
            >
                <div className='relative flex items-center justify-center '>
                    <p className='block w-max font-semibold text-gray-900 text-xs py-1 '>
                        {text}
                    </p>
                </div>
            </Button>
        </div>
    );
};

export default SingleInterest;
