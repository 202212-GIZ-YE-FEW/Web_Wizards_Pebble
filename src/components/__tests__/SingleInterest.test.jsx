import renderer from "react-test-renderer";

import SingleInterest from "../SingleInterest";

it("renders correctly", () => {
    const tree = renderer.create(<SingleInterest />).toJSON();
    expect(tree).toMatchSnapshot();
});
