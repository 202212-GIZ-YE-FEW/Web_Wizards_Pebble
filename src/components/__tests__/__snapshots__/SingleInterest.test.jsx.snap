// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
>
  <button
    className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white !border-2 !rounded-lg !shadow-[2px_2px_#1A1A1A]"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="relative flex items-center justify-center "
    >
      <p
        className="block w-max font-semibold text-gray-900 text-xs py-1 "
      />
    </div>
  </button>
</div>
`;
